system/core/pages.php

```diff
@@ -47,17 +47,57 @@ abstract class HyphaSystemPage extends HyphaPage {
 		abstract class for a page backed by datatype and xml file.
 	*/
 	abstract class HyphaDatatypePage extends HyphaPage {
+		const INDEX_TABLE_COLUMNS_TITLE = 'title';
+		const INDEX_TABLE_COLUMNS_AUTHOR = 'author';
+		const INDEX_TABLE_COLUMNS_DATE = 'date';
+		const INDEX_TABLE_COLUMNS_STATUS = 'status';
+
 		public $pageListNode, $language, $pagename, $privateFlag;
 
 		function __construct(HyphaDomElement $node, RequestContext $O_O) {
 			parent::__construct($O_O);
 			$this->replacePageListNode($node);
 		}
 
+		/**
+		 * @return string
+		 */
 		public static function getDatatypeName() {
 			return str_replace('_', ' ', get_called_class());
 		}
 
+		/**
+		 * @return array
+		 */
+		public static function getIndexTableColumns() {
+			return [
+				__(self::INDEX_TABLE_COLUMNS_TITLE),
+				__(self::INDEX_TABLE_COLUMNS_AUTHOR),
+				__(self::INDEX_TABLE_COLUMNS_DATE),
+			];
+		}
+
+		/**
+		 * @return array
+		 */
+		public function getIndexData() {
+			$id = $this->pageListNode->getAttribute('id');
+			$date = $this->getSortDateTime();
+			$dataMtx = [
+				__(self::INDEX_TABLE_COLUMNS_TITLE) => [
+					'class' => 'type_'.get_class($this).' '.($this->privateFlag ? 'is-private' : 'is-public'),
+					'sort' => preg_replace("/[^A-Za-z0-9]/", '', $this->getTitle()).'_'.$id,
+					'value' => '<a href="'.$this->language.'/'.$this->pagename.'">'.$this->getTitle().'</a>',
```

Comment:
Should this be `html_value` to distinguish between html and text value? Alternatively, the non-HTML values should be `htmlspecialchars`'d? Alternatively, this could maybe build a `DomNode` to indicate that it is already HTML (combined with my other suggestion to build a DOM tree instead of an HTML string), and then any strings can be treated as text (and thus encoded).